crates/core/src/host/host_controller.rs

```diff
@@ -216,36 +241,33 @@ impl HostController {
     pub async fn spawn_module_host(&self, module_host_context: ModuleHostContext) -> Result<ModuleHost, anyhow::Error> {
         let key = module_host_context.dbic.database_instance_id;
 
-        let (module_host, start_module, start_scheduler) =
-            Self::make_module_host(module_host_context, self.energy_monitor.clone())?;
+        let (module_host, start_scheduler) = self.make_module_host(module_host_context)?;
 
-        let old_module = self.modules.lock().unwrap().insert(key, module_host.clone());
+        let old_module = self.modules.lock().insert(key, module_host.clone());
         if let Some(old_module) = old_module {
             old_module.exit().await
         }
-        start_module.start();
+        module_host.start();
         start_scheduler.start(&module_host)?;
 
         Ok(module_host)
     }
 
-    fn make_module_host(
-        mhc: ModuleHostContext,
-        energy_monitor: Arc<dyn EnergyMonitor>,
-    ) -> anyhow::Result<(ModuleHost, ModuleStarter, SchedulerStarter)> {
+    fn make_module_host(&self, mhc: ModuleHostContext) -> anyhow::Result<(ModuleHost, SchedulerStarter)> {
         let module_hash = hash_bytes(&mhc.program_bytes);
-        let (module_host, module_starter) = match mhc.host_type {
+        let (threadpool, energy_monitor) = (self.threadpool.clone(), self.energy_monitor.clone());
+        let module_host = match mhc.host_type {
             HostType::Wasmer => {
                 // make_actor with block_in_place since it's going to take some time to compute.
                 let start = Instant::now();
                 let actor = tokio::task::block_in_place(|| {
                     wasmer::make_actor(mhc.dbic, module_hash, &mhc.program_bytes, mhc.scheduler, energy_monitor)
                 })?;
-                log::trace!("wasmer::make_actor blocked for {}us", start.elapsed().as_micros());
-                ModuleHost::spawn(actor)
+                log::trace!("wasmer::make_actor blocked for {:?}", start.elapsed());
```

Comment:
Why is this debug now?

Reply:
it does the same thing (when the Duration is in the microseconds), I didn't see a reason to do our own formatting when the Duration Debug impl works fine